GoogleForCreators/web-stories-wp
# Media: Video Cropping Prototype

## Context

As per https://github.com/GoogleForCreators/web-stories-wp/issues/5985,  implement video cropping to cut off-screen parts of videos



## Summary

Adds cropping for off canvas top, right, bottom, left i.e. hidden or off-canvas video

https://www.youtube.com/watch?v=P9Co7r7Droo



## Relevant Technical Choices



## To-do





## User-facing changes



## Testing Instructions

Created [this video](https://user-images.githubusercontent.com/62242/187041498-8d192f78-e376-4761-8ae1-0a85f364a9b0.mp4) - each square is 80px X 80px








This PR can be tested by following these steps:

1. Enable the experiment for `Enable support for cropping cut off-screen parts of videos`
2. Add the video above to the canvas
3. Position the video partially off canvas
4. Select `Crop off-screen video parts` from the Right Click Menu
5.  The video should now be cropped with the off cavas part of the video removed.


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #12093
